src/legacy/server/index_patterns/service/lib/resolve_time_pattern.ts

```diff
@@ -34,15 +36,16 @@ import { callIndexAliasApi } from './es_api';
  *                            and the indices that actually match the time
  *                            pattern (matches);
  */
-export async function resolveTimePattern(callCluster, timePattern) {
+export async function resolveTimePattern(callCluster: APICaller, timePattern: string) {
   const aliases = await callIndexAliasApi(callCluster, timePatternToWildcard(timePattern));
 
-  const allIndexDetails = chain(aliases)
-    .reduce((acc, index, indexName) => acc.concat(
-      indexName,
-      Object.keys(index.aliases || {})
-    ), [])
-    .sort()
+  const allIndexDetails = chain<string[]>(aliases as string[])
```

Comment:
I think the `string[]` type is actually wrong.  Could you go into the `es_api` file where `callIndexAliasApi` is defined and have it return `Promise<IndicesAliasResponse>` with types:
```
export interface IndicesAliasResponse {
  [index: string]: IndexAliasResponse;
}

export interface IndexAliasResponse {
  aliases: {
    [aliasName: string]: Record<string, any>;
  };
}
```

Then here do:

```
chain<IndicesAliasResponse>
```

Reply:
Great. Thanks.